From cdbf84f20295c8a78624318aa6fdfff3f5c8ce27 Mon Sep 17 00:00:00 2001
From: Keith Packard <keithp@keithp.com>
Date: Thu, 11 Jun 2009 20:57:09 +0000
Subject: intel_batch_init: test have_gem, not directRenderingType

Under KMS, directRenderingType will get set to DRI_NONE during driver
initialization. When the first batch buffer is allocated, as
directRenderingType is DRI_NONE, the GEM bufmgr would get trashed as
intel_batch_init called a fake-bufmgr specific function.

Signed-off-by: Keith Packard <keithp@keithp.com>
---
diff --git a/src/i830_batchbuffer.c b/src/i830_batchbuffer.c
index a6d9c6e..c24409b 100644
--- a/src/i830_batchbuffer.c
+++ b/src/i830_batchbuffer.c
@@ -135,7 +135,7 @@ intel_batch_init(ScrnInfoPtr pScrn)
 
     intel_next_batch(pScrn);
 
-    if (pI830->directRenderingType <= DRI_NONE) {
+    if (!pI830->have_gem) {
 	if (IS_I830(pI830) || IS_845G(pI830)) {
 	    intel_bufmgr_fake_set_exec_callback(pI830->bufmgr,
 						intel_nondrm_exec_i830,
--
cgit v0.8.2
